src/app/TransactionModal.tsx

```diff
@@ -12,31 +13,45 @@ type Props = {
   setError: (error: string) => void;
   writeQueue: NamedTransaction[];
   setWriteQueue: (fn: NamedTransaction[]) => void;
+  currentTxn?: NamedTransaction;
+  setCurrentTxn: (fn: NamedTransaction | undefined) => void;
 };
 
 function TransactionModal(props: Props) {
   const [hadHash, setHadHash] = useState(false);
   const [hash, setHash] = useState<`0x${string}`>();
   const [error, setError] = useState<string>();
   const publicClient = usePublicClient();
-  const [currentTxn, setCurrentTxn] = useState<NamedTransaction>();
+  const [processingIndexNumber, setProcessingIndexNumber] = useState<number>(0);
+  useState<boolean>(false);
 
   function closeModal() {
     props.setIsOpen(false);
     if (error) {
       setHash(undefined);
     }
     setError("");
+
+    if (props.writeQueue[props.writeQueue.length - 1]?.status === "succeeded")
+      resetTransactionStatus();
   }
 
   function openModal() {
     props.setIsOpen(true);
   }
 
+  function resetTransactionStatus() {
+    props.setWriteQueue([]);
+    props.setCurrentTxn(undefined);
+
+    setProcessingIndexNumber(0);
+    setHash(undefined);
+  }
+
   useEffect(() => {
     if (!props.isOpen && !hadHash && hash) {
       setHadHash(true);
-      openModal();
+      // openModal();
```

Comment:
if we don't need this anymore, let's delete this

Reply:
removed it along with the hadHash state hook